index.js

```diff
@@ -199,7 +200,16 @@ const fastifyReplyFrom = fp(function from (fastify, opts, next) {
       } else {
         copyHeaders(rewriteHeaders(res.headers, this.request), this)
       }
-      this.code(res.statusCode)
+      try {
+        this.code(res.statusCode)
+        this.request.log.warn(err, 'response has invalid status code')
+      } catch (err) {
+        if (err.code === 'FST_ERR_BAD_STATUS_CODE') {
+          onError(this, { error: new BadGatewayError() })
+        } else {
+          onError(this, { error: new InternalServerError(err.message) })
+        }
```

Comment:
Should this log be in the catch? If `this.code(res.statusCode)` succeeds it will still log
```suggestion
      } catch (err) {
        if (err.code === 'FST_ERR_BAD_STATUS_CODE') {
          onError(this, { error: new BadGatewayError() })
        } else {
          onError(this, { error: new InternalServerError(err.message) })
        }
         this.request.log.warn(err, 'response has invalid status code')
```